### PR TITLE
Game search dashboard

### PR DESCRIPTION
**Pull request by rtdickey**

### CODE REVIEW

File: src/pages/protected/Dashboard.tsx [hunk 1]
```diff
@@ -1,11 +1,40 @@
...
+} from "../../components/boardgame-geek";
+import { useSearchGamesByNameQuery } from "../../redux/services/gameSearch";
 
 const Dashboard = () => {
+  const [searchResult, setSearchResult] = useState("");
+
+  const handleOnSubmit = (value: string) => {
+    setSearchResult(value);
+  };
+  const { data, error, isLoading } = useSearchGamesByNameQuery(searchResult);
```
Comment (Genide) [inline src/pages/protected/Dashboard.tsx:15]: Does this cause you to make a call to the api before you even type in anything to search?

**Comment by rtdickey** [comment]
@Genide I'm not sure if this is what you had in mind, but I utilized RTKQ's mutation to call it's trigger inside the form submit. This allowed me to get around the hook restriction.

File: api/hot/index.ts [hunk 2]
```diff
@@ -16,9 +16,9 @@ const httpTrigger: AzureFunction = async function (
   const responseXml = await (
     await fetch(`https://api.geekdo.com/xmlapi2/hot`)
   ).text();
-  context.log(responseXml);
+  // context.log(responseXml);
```
Comment (kevinrs5855) [inline api/hot/index.ts:19]: we just log debug or remove comments

File: api/thing/index.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    return;
+  }
+
+  // 1543,5256,1542
+  const responseXml = await (
+    await fetch(
+      `https://api.geekdo.com/xmlapi2/thing?id=${searchParameter}&type=boardgame`
+    )
+  ).text();
+  // context.log(responseXml);
```
Comment (kevinrs5855) [inline api/thing/index.ts:30]: We should remove commented out code

**Review by kevinrs5855 — APPROVED** [review]
Looks good, one issue, the search api endpoint and thing api endpoint are accessible by users not logged in, so they could technically over use the APIs.

**Pull request merged**